tools/basename_picker.cwl

```diff
@@ -0,0 +1,36 @@
+cwlVersion: v1.2
+class: ExpressionTool
+id: basename_picker
+requirements:
+  - class: InlineJavascriptRequirement
+
+inputs:
+  read1_filename: string
+  output_basename: 'string?'
+  sample_name: 'string?'
+  star_rg_line: 'string?'
+
+outputs:
+  outname:
+    type: string
+  outsample:
+    type: string
+  outrg:
+    type: string
+
+expression: |
+  ${
+    var name = inputs.output_basename ? inputs.output_basename : inputs.read1_filename;
+    var sample = inputs.sample_name ? inputs.sample_name : inputs.read1_filename;
```

Comment:
If the reads named in the standard illumina format, it might be nice to strip the information appended at the end of the filename. For example if R1 is "SampleName_S1_L001_R1_001.fastq.gz" just use "SampleName"

Reply:
Also note that the lane number (L001) may not be present if --no-lane-splitting was provided to bcl2fastq, so account for that in your regular expression if you decide to implement this

Reply:
While this is would be a nice to have, it goes beyond the scope of this task/PR, and I will not be implementing it here.